internal/manifests/collector/annotations.go

```diff
@@ -40,14 +41,20 @@ func Annotations(instance v1alpha1.OpenTelemetryCollector) map[string]string {
 			annotations[k] = v
 		}
 	}
+
+	hash, err := getConfigMapSHA(instance.Spec.Config)
+	if err != nil {
+		return nil, err
+	}
+
 	// make sure sha256 for configMap is always calculated
-	annotations["opentelemetry-operator-config/sha256"] = getConfigMapSHA(instance.Spec.Config)
+	annotations["opentelemetry-operator-config/sha256"] = hash
 
-	return annotations
+	return annotations, nil
 }
 
 // PodAnnotations return the spec annotations for OpenTelemetryCollector pod.
-func PodAnnotations(instance v1alpha1.OpenTelemetryCollector) map[string]string {
+func PodAnnotations(instance v1alpha2.OpenTelemetryCollector) (map[string]string, error) {
```

Comment:
note: in the future, i think we should make the getConfigMapSHA a method on the collector itself that returns no error. Note worth doing in this PR, but doing so in the future would help us clean up a lot of the error checking this introduces.

Reply:
I made an issue for this:
- https://github.com/open-telemetry/opentelemetry-operator/issues/2591